### DISCUSSION TITLE
Change gist name and description

### DISCUSSION THREAD

**Discussion by ArwynFr** [community/community#13772] [Other Feature Feedback, Questions, & Ideas]
Currently gists always take the name of the first file based on ASCII sorting, which does not always make sense. I would like to be able to give an independent name to my gists, not based on the filename, and update it after creation.

Also, you can add a description to the gist but this cannot be updated afterwards. I also would like to change the gist description, rather than deleting and creating the gist again.

**Comment by queenofcorgis** [comment]
Thanks for this feedback and why it is important to you!

**Reply by franzwong** [reply]
Can we make the title more obvious? If I have multiple gist with same file names (e.g. README.md), it is difficult to find the one I want. Or does gist encourage people putting title as file name?

**Reply by Deborahqp** [reply]
wow this very bad(

**Reply by martin-braun** [reply]
Gists are just repositories. I want to create Gists instead of fully fledged repositories on GitHub for simple things that have 3 files max. I enjoy committing changes like on regular repositories, but that means that I need a `.gitignore` to work with it in a sane manner. Well `.gitignore` will always come first and thus it messes with the presentation of my Gists. 

<img width="602" alt="image" src="https://github.com/community/community/assets/24259245/172a2f92-1c49-4ba3-94ec-af956423904f">

This sucks, please fix that problem. Simply allow us to mark a file as default file.

**Comment by radar-blip** [comment]
You can change the description by clicking on the `Edit` button at top of page.

**Comment by jfullstackdev** [comment]
I just check it now, 

for example, when I did not enter anything in the title, it will generate the same name 

![image](https://user-images.githubusercontent.com/47092464/201509006-efbd96cd-6a94-4de2-b7c2-ffc18a69e041.png)

but the unique is the URL path

![image](https://user-images.githubusercontent.com/47092464/201509023-43c47e27-3e2b-4060-9d27-f30c6cf7b58a.png)

that's also one of the requests, unfriendly file names when you send the link to someone

but how do you use it ? because when you embed it, the URL will not be noticed because it's simply embedded

**Comment by Len42** [comment]
Yes, please let us name our gists! Arbitrarily choosing one of the filenames as the gist name often gives the gist a misleading or uninformative name.

**Comment by tkrotoff** [comment]
FYI This issue has been opened since May 16, 2014: https://github.com/isaacs/github/issues/194

Check also these stackoverflow questions: https://stackoverflow.com/q/19896900, https://stackoverflow.com/q/17665267

**Comment by pseusys** [comment]
Maybe we should file an issue to official GitHub support, if it's not been done already?
Or, it it has, could you please post a link here, I would definitely give it a 👍.

**Comment by dotysan** [comment]
The workaround of adding a ' The Title' file is not ideal. The first file is also used on your gist homepage for a summary. And with the leading space, the summary isn't shown. Instead you get an error message "We couldn’t find that file to show." An example, currently at the top of mine. https://gist.github.com/dotysan
![image](https://github.com/community/community/assets/5060170/5c935a2d-6b07-49ec-8621-4d9f361e4ddb)

**Reply by coralkashri** [reply]
A Similar workaround is to name a `.md` file with a name starting with "A": 
![image](https://github.com/community/community/assets/41470875/12d81141-d903-4514-b0db-bb718308e117)
https://gist.github.com/coralkashri/e47b27d427b8f90226e280fa9ac4c9fc/88cd598e624113dc35f50275d3dd86560c6c37a7

**Comment by Killerzbie** [comment]
> Can we make the title more obvious? If I have multiple gist with same file names (e.g. README.md), it is difficult to find the one I want. Or does gist encourage people putting title as file name?

**Comment by ArwynFr** [comment]
Here is how I workaround this problem now: I create a file named `!!README.adoc` with a description of my gist. The file gets displayed as a preview and the primary header is larger than the gist name. I stop looking for the gist name and use that title instead. It looks like this:

![image](https://github.com/community/community/assets/10365998/c54b1b19-7d7b-4f41-9da0-65b135ac0834)

Note: It only displays few first lines of the document, be concise on the description. Probably works with markdown as well, but not tested.

**Comment by GiwbyAlbatross** [comment]
Is this solved? We can now rename gists via the `Edit` button.

**Reply by ArwynFr** [reply]
> Is this solved? We can now rename gists via the `Edit` button.

No, this field acts more like a description.
The text appearing next to the username is still a filename.

Here is how it looks:
<img width="1200" height="246" alt="image" src="https://github.com/user-attachments/assets/02b2bbd3-ca7d-4796-83b5-1415c4dd3b4f" />

**Comment by ernstki** [comment]
It's a shame, because for smallish projects, Gists are Just Right™, whereas a full GitHub repo brings all kinds of unnecessary baggage and distractions.

But with a "real" GitHub repository, you don't have to do weird workarounds to have a README or a `.gitignore` file, and `.gitattributes` works as expected.

**Comment by andig** [comment]
Same here. This is so obvious that it hurts it's still open after almost 12 years... Anybody at GH listening?

**Reply by martin-braun** [reply]
They won't listen and they give a shit. Here is how you circumvent their limitation, an example: https://gist.github.com/martin-braun/e10b7cafdd55feea44bc2d06d5c2c972

(You just rename your `README.md` to `!project-name.md`, which will put your documentation on top and name the gist after your documentation file.)

**Reply by ernstki** [reply]
@martin-braun Great tip, and it totally works fine.

Don't lose hope. Some kind-hearted individual GitHub engineer might still go out of their way to fix this as a side project.

At the organizational level, though, GitHub has reportedly been [subsumed by Microsoft's CoreAI division](https://www.techradar.com/pro/github-ceo-resigns-is-this-the-latest-sign-of-its-microsoft-absorption), and [its focus is elsewhere right now](https://thenewstack.io/github-will-prioritize-migrating-to-azure-over-feature-development/): scaling out and training their money-makers on all the code we wrote for them.

**Reply by PaulRBerg** [reply]
Another stop-gap solution is to create a dummy, empty file at the top of the gist with a custom name.
